### ISSUE TITLE
TKB Home wall thermostat TZE93.716 on HomeyPro (2023)

### ISSUE THREAD

**Issue by SteffenM2023** [pshotton/com.tkbhome#2]
(via Google translator) 
Hello,
I just switched from Homee to HomeyPro (2023). I use some TKB Home room thermostats to control underfloor heating actuators. Unfortunately, the TKB Home wall thermostat TZE93.716 (I have 10 of them) is not supported in the TKB Home app. Is it possible to integrate this? Unfortunately I have no experience with Homey. Thanks in advance. 
Steffen

**Comment by pshotton** [comment]
Hi Steffen. I'll do some research and see. I don't have access to one of these so I'll need your help for testing :)
Can you add it as a basic ZWave device? if so can you let me know the manufacturer Id, product Id and productType Id from the settings page?
Thanks
Phil

**Comment by Infiniverse** [comment]
Hi. I’ve got lots of TZE93 devices too. Just bought a homey pro. Are you able to help get these supported? I’m a programmer so can help in any way you need.

**Comment by Infiniverse** [comment]
I’ve got a homey pro and need to get my TKHome thermostats working, so if necessary I’ll need to understand how this code works. Hopefully Phil is still available to help.

**Comment by SteffenM2023** [comment]
Hello, my English is not so good, I work with Google Translate.

Einstellungen

Energie:  Immer an
Energieverbrauch wenn aus (Watt): 0
Energieverbrauch wenn an (Watt): 0

Z-Wave-Verbindungen
Gruppe 1: 1

Z-Wave-Raw-Konfigurationsparameter
id, size, value; id, size, value; ...

Z-Wave-Geräteinformation
Firmware-Version: 2
Firmware-Unterversion: 3
Firmware-ID: 5381
Hardware-Version: 1
Sicher: ⨯
Batterie/Akku: ⨯
Node ID: 82
Hersteller-ID: 280
Produkttyp-ID: 785
Produkt-ID: 1285
Device Class (Basic): BASIC_TYPE_ROUTING_SLAVE
Device Class (Generic): GENERIC_TYPE_THERMOSTAT
Device Class (Specific): SPECIFIC_TYPE_THERMOSTAT_GENERAL_V2

Thank You
![Thermostat](https://github.com/pshotton/com.tkbhome/assets/134263702/75f231a7-5192-4d8b-b830-f014eb76c1e1)

**Comment by pshotton** [comment]
I've been absent for a while as work took a time-consuming turn, but I'm
back online now and looking to work through the outstanding items.
Stefan, thanks for the information. This should give me enough to rough out
a module for the TZE93.

If you or Josef could loan me a TZE93 module I would appreciate it!

Thanks
Phil

**Comment by SteffenM2023** [comment]
Hello, 

thank you for the message. I also had too much to do. I am happy to provide a thermostat. I come from Germany, so I would have to research whether shipping is easy. Where do you live?

Steffen

**Comment by pshotton** [comment]
Stefan, I'm in England. I've looked online and the thermostat doesn't seem
to be available.
I can probably write the code, there seems to be quite a bit of information
online, but I won't be able to test it.
Give me a few days and I'll try to implement support in the beta version of
the TKB app.

Thanks
Phil

**Comment by SteffenM2023** [comment]
Of course, I would be happy about that. I also looked and didn't find one in a shop, and there would be taxes.

Thanks
Steffen

**Comment by pshotton** [comment]
Version 3.1.0 published to test here
https://homey.app/en-gb/app/com.tkbhome/TKB-Home/test/
Contains initial support for TZE93.
Please test and log any issues.

Thanks
Phil

**Comment by SteffenM2023** [comment]
Hello,

I installed the TZB93 thermostat, went without errors. Setting the temperature and displaying it works. I haven't found an error yet. Where can I look for it? I can create a diagnostic report, but I only get one digit number.

Thanks in advance for the very quick support - amazing!

What and how should I test further?

Greetings Steffen
![TZE93 - 1](https://github.com/pshotton/com.tkbhome/assets/134263702/1c0a992f-e242-42bd-b118-0e4bca77fc83)
![TZE93 - 2](https://github.com/pshotton/com.tkbhome/assets/134263702/2dca9ee1-3c06-4ca3-8b36-177306a90e1a)

**Comment by pshotton** [comment]
Thanks for trying it out Stefan. Looks like it all works fine, I shall push
the test version into production.

**Comment by pshotton** [comment]
Test release has been promoted to live. Enjoy!
